src/main/java/io/jenkins/plugins/intotorecorder/transport/Grafeas.java

```diff
@@ -0,0 +1,90 @@
+/**
+ *
+ */
+package io.jenkins.plugins.intotorecorder.transport;
+
+import io.github.in_toto.models.Link;
+import java.net.URI;
+import java.util.*;
+import com.google.gson.Gson;
+
+import java.io.IOException;
+
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpResponse;
+import com.google.api.client.http.ByteArrayContent;
+import com.google.api.client.http.GenericUrl;
+
+
+public class Grafeas extends Transport {
+
+    URI uri;
+    GrafeasOccurrence occurrence;
+
+    public class GrafeasOccurrence {
+        public String name;
+        public String noteName;
+        public Map<String, String> resource = new HashMap<String, String>();
+        public String kind = "INTOTO";
+        public Link intoto;
+
+        public GrafeasOccurrence(String name, String noteName, String resourceUri) {
+            this.name = name;
+            this.noteName = noteName;
+            this.resource.put("uri", resourceUri);
+        }
+    }
+
+    private static Map<String, String> getParameterMap(String parameterString) {
+        String[] items = parameterString.split("&");
+        Map<String, String> parameterMap = new HashMap<String, String>();
+        for (String item : items) {
+            String[] pair = item.split("=");
+            parameterMap.put(pair[0], pair[1]);
+        }
+        return parameterMap;
+    }
+
+    public Grafeas(URI uri) {
+        this.uri = uri;
+
+        String parameterString = uri.getQuery();
+
+        Map<String, String> parameterMap = this.getParameterMap(parameterString);
+
+        GrafeasOccurrence occurrence = new GrafeasOccurrence(
+            parameterMap.get("name"),
+            parameterMap.get("noteName"),
+            parameterMap.get("resourceUri")
+        );
+
+        this.occurrence = occurrence;
+    }
+
+    public void submit(Link link) {
+        this.occurrence.intoto = link;
+
+        Gson gson = new Gson();
+        String jsonString = gson.toJson(this.occurrence);
+
+        String destination = this.uri.toString().split("\\?")[0].substring("grafeas+".length());
+
+        // FIXME: Shamelessly copied from GenericCRUD.java
```

Comment:
This bit needs to go in a `common` file somewhere or I need to import it from GenericCRUD (?).

Reply:
Wouldn't it make sense to inherit from GenericCRUD?

Reply:
I'll figure this out, I don't see why not. Do we have any organizational qualms about one transport inheriting from another instead of the default base?

Reply:
I wouldn't know why. It still inherits from `Transport`, but there would be a class in-between, i.e. `GenericCRUD` that provides almost exactly the desired behavior. What organizational qualms did you have in mind?

Reply:
Basically a hierarchy of the transport classes, but I think it doesn't matter. I'll do this.

Reply:
Update: based on our discussion, I've currently retained it as it and I'd like to open an issue to track the repetition of the HTTP/HTTPS submit code across `GenericCRUD`, `Etcd`, and now `Grafeas`. But, I think they're disabled for this repository. cc @SantiagoTorres

Reply:
I think inherriting from GenericCRUD would probably be ideal (and it rather matches the functional hierarchy of these transports). I imagine you'd have to work a little bit of the bits and pieces of the GenericCRUD transport so it can be easily inheritted. If that's too much hassle for this PR, then I'm ok with merging then re-factoring in the near future.

Reply:
If I understand correctly, @lukpueh and I noticed that the `Etcd` transport also uses similar code for HTTP/HTTPS requests and also needs to be part of the refactor, and so I figured I'd open an issue and we could merge this one. Except, I think this repo doesn't have issues enabled.

And yes, `GenericCRUD` needs some work for it to be inheritable.

Reply:
Though, we should keep in mind that this depends on the Grafeas PR, so while we leave it open, it's probably okay to make the changes here.

Reply:
I think for the sake of moving forward we can have this and then have a refactoring PR that simplifies the class hierarchy

Reply:
Gotcha. I'm resolving this in the meantime on this thread.